File: stack.py
class Stack(object):

    def __init__(self):
        self.stack = []
        self.stack_len = 0

    def push(self, n):
        self.stack.append(n)
        self.stack_len += 1

    def pop(self):
        if self.is_empty():
            print 'Stack Empty'
        else:
            self.stack_len -= 1
            print 'pop({})'.format(self.stack.pop(self.stack_len))

    def is_empty(self):
        if self.stack_len == 0:
            return True
        return False

stack = Stack()
stack.push(1)
stack.push(2)
stack.pop()
stack.pop()
stack.pop()

print stack.stack

File: stack.rb
class Stack
	def initialize
		@stack = []
		@count = 0
	end
	def push(n)
		@stack.push(n)
		@count += 1
	end
	def pop
		unless is_empty
			@stack.pop
			@count -= 1
		end
	end
	def stack
		return @stack
	end
	def count
		return @count
	end
	def is_empty
		return true if @count == 0
		return false
	end
	
end

stack = Stack.new
stack.push(2)
stack.push(3)
print stack.stack
stack.pop()
print stack.stack
